fdbserver/VersionIndexerInterface.actor.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * VersionIndexerInterface.actor.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#if defined(NO_INTELLISENSE) && !defined(FDBSERVER_VERSION_INDEXER_INTERFACE_ACTOR_G_H)
+#define FDBSERVER_VERSION_INDEXER_INTERFACE_ACTOR_G_H
+#include "fdbserver/VersionIndexerInterface.actor.g.h"
+#elif !defined(FDBSERVER_VERSION_INDEXER_INTERFACE_ACTOR_H)
```

Comment:
Sorry, why FDBSERVER_VERSION_INDEXER_INTERFACE_ACTOR_G_H is not defined in this case?

Reply:
This is a crappy preprocessor-hack that we use all over the place (it's super hard to read and I usually just copy&paste this block from another `.actor.h` file and modify it).

When we compile this we pass `NO_INTELLISENSE` and when we use `clangd` we don't. So without `NO_INTELLISENSE` defined, we never include the `actor.g.h` file and only execute the above. If `NO_INTELLISENSE` is defined, if will include the generated file which in turn will define `FDBSERVER_VERSION_INDEXER_INTERFACE_ACTOR_H` (so the non-actorcompiled code willl not be included at all). 

Basically, if you stare at this for a very long time you will get it :D